pkg/cache/v2/simple.go

```diff
@@ -366,8 +378,65 @@ func createResponse(request *Request, resources map[string]types.ResourceWithTtl
 	}
 }
 
+// CreateDeltaWatch returns a watch for a delta xDS request which implements the Simple SnapshotCache.
 func (cache *snapshotCache) CreateDeltaWatch(request *DeltaRequest, st *stream.StreamState) (chan DeltaResponse, func()) {
-	return nil, nil
+	nodeID := cache.hash.ID(request.Node)
+	t := request.GetTypeUrl()
+
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	info, ok := cache.status[nodeID]
+	if !ok {
+		info = newStatusInfo(request.Node)
+		cache.status[nodeID] = info
+	}
+
+	// update last watch request times
+	info.mu.Lock()
+	info.lastDeltaWatchRequestTime = time.Now()
+	info.mu.Unlock()
```

Comment:
it might be nice to have these simple getters use a helper function, e..g

```
func (i *info) SetLastDeltaWatchRequestTime(t time.Time) {
  i.Lock()
  defer i.Unlock()
  return i.lastDeltaRequestTime = t
}
```

as this avoids leaking the need for a lock to the caller

Reply:
Yup, I'll change that. I just noticed the code dupe